group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1350,6 +1351,11 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
                         .withMembers(group.members())
                         .withStaticMembers(group.staticMembers())
                         .withSubscriptionMetadata(subscriptionMetadata)
+                        .withSubscriptionType(ConsumerGroup.subscriptionType(
+                            subscribedTopicsMemberCount,
+                            group.numMembers(),
+                            group.subscriptionType()
+                        ))
```

Comment:
I figured we don't need to compute it if it's not necessary since we aren't updating the actual value anyways. It is only required in this if block right